### ISSUE TITLE
How to generate the bearer token for authorization?

### ISSUE THREAD

**Issue by abhijeet-ghosh** [cowinapi/developer.cowin#177]
Can someone please help me with the process to generate the Authorization token needed to use the API. Thanks !

**Comment by deepgandhi** [comment]
For public api use:- https://apisetu.gov.in/public/marketplace/api/cowin#/User%20Authentication%20APIs

For protected api use in production server:-
https://apisetu.gov.in/public/marketplace/api/cowin/cowin-protected-v2#/Authentication%20APIs

But for protected api you need to have "x-api-key" of production server, which is not available. 

I have a workaround. Generate and Confirm OTP on cowin.gov.in and get the token from session data. Use that token for Bearer Authorization.

**Comment by siddhesh1770** [comment]
> Can someone please help me with the process to generate the Authorization token needed to use the API. Thanks !

Token ID is generated on successful OTP submission
refer to - [https://apisetu.gov.in/public/marketplace/api/cowin/cowin-public-v2#/User%20Authentication%20APIs/confirmOTP](url)

**Comment by shivam-methi** [comment]
You can Generate Token, for which you need to do is:
1. use this API: /v2/auth/public/generateOTP from here you will get the **txnId**
2. use this: https://emn178.github.io/online-tools/sha256.html & enter your otp received on mobile and from here you will get 
   converted SHA 256 data which will be your **otp** field.
3. Use this API: /v2/auth/public/confirmOTP to generate the token and can use 2 parameters generated from 1 & 2.

**Comment by abhijeet-ghosh** [comment]
Thanks everyone for the response. I am able to generate the token, but when I am trying to run the below API, I am getting error message as "TypeError: NetworkError when attempting to fetch resource.". Not sure of the error.  Please help.

API Used: /v2/appointment/sessions/public/findByPin

**Comment by gopal-amlekar** [comment]
Sometimes using user-agent header works.

'user-agent': Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36

But what do you do with the bearer token you obtain. Booking the appointment isn't possible anyway without captcha.

**Comment by deepgandhi** [comment]
You can use it for other api. You have to pass it on headers "Authorization":"Bearer **token**". Replace the **token** with one you generated.

**Comment by me-nkr** [comment]
@shivam-test the first api return {"message":"Forbidden"}

**Comment by roopalrjain** [comment]
> You can Generate Token, for which you need to do is:
> 
>     1. use this API: /v2/auth/public/generateOTP from here you will get the **txnId**
> 
>     2. use this: https://emn178.github.io/online-tools/sha256.html & enter your otp received on mobile and from here you will get
>        converted SHA 256 data which will be your **otp** field.
> 
>     3. Use this API: /v2/auth/public/confirmOTP to generate the token and can use 2 parameters generated from 1 & 2.

Yes this workaround will work to get token, but remember that token is valid only for 5 mins, it will expire in 5 min if session is idel for continues 5 min, and max expiry time is 15 min, so if you are actively using that session then max you can use it for 15 mins.

**Comment by kprageet** [comment]
Could some with bearer token generation url for public apis.  Look like /v2/auth/public/generateOTP  return "Forbidden". Wondering toke needs passed as part for the header?
